src/light-signup.js

```diff
@@ -1,13 +1,19 @@
+import defaultsDeep from 'lodash/object/defaultsDeep';
+
+const defaultOptions = {
+	signupUrl: '/signup/api/light-signup'
+};
+
 export default {
 
-	init() {
+	init(el, options = {}) {
+		defaultsDeep(options, optionsFromData(el), defaultOptions);
 
-		const lightSignupComponent = document.querySelector('.n-light-signup');
-		const closeButton = document.querySelector('.n-light-signup__close');
-		const lightSignupForm = document.querySelector('.n-light-signup__form');
-		const displaySection = document.querySelector('.n-light-signup__secondary');
-		const emailField = document.querySelector('.n-light-signup__email');
-		const invalidEmailMessage = document.querySelector('.n-light-signup__email-error-msg');
+		const closeButton = el.querySelector('.o-light-signup__close');
```

Comment:
As per the spec, the JS should use data attributes to hook into the DOM and classes should be used for styling